### PR TITLE
Mention life-cycle API in Request assertions recipe

### PR DESCRIPTION
**Pull request by Izhaki**
resolves https://github.com/mswjs/msw/issues/719

### CODE REVIEW

File: docs/recipes/request-assertions.mdx [hunk 1]
```diff
@@ -19,6 +19,12 @@ Mock Service Worker gives you a guarantee over network communication.
 
 Your application fetches data for a reason, so prefer testing that reason as the outcome of the request. If a request's method, headers, or body are invalid, your application is likely to get an invalid response back. Below we are going to take a look at some recommended strategies to request testing.
 
+### Special cases
+
+There are, however, some special cases where asserting on network requests is the only option. These include, for example, polling, where no other side effect can be asserted upon.
+
+For such cases, Mock Service Worker allows tracking of network requests through the [Life-cycle events API](https://mswjs.io/docs/extensions/life-cycle-events#tracking-a-request).
```
Comment (kettanaito) [inline docs/recipes/request-assertions.mdx:26]: Could you please use a relative path to the page in the docs?

```suggestion
For such cases, Mock Service Worker allows tracking of network requests through the [Life-cycle events API](/docs/extensions/life-cycle-events#tracking-a-request).
```

Thanks!

**Review by kettanaito — CHANGES_REQUESTED** [review]
Hey, @Izhaki. Thank you for adding this!
Please, could you fix the link to the page in the docs?

**Review by kettanaito — CHANGES_REQUESTED** [review]
Do you think it'd make more sense if we moved this section to the very bottom of the page?
I think that the order in which we suggest the approaches matters, and "special cases" is the last resort if any other suggestion is not applicable. We still want the readers to go through those suggestions to assess their relevance.

**Comment by Izhaki** [comment]
@kettanaito For me, I was looking for a way to assert on requests, which is how I ended up looking at the `Request assertions` section.

The content under `Request assertions` makes it pretty clear you shouldn't assert on requests.

The following sections - `Include failure scenarios` and `Treat unhandled requests` discuss other best practices. So at that point the reader might conclude asserting on requests is not possible, and it will take scrolling to reveal otherwise.

So for me, `Special cases` is tightly linked to the `Request assertions` heading and the text under it.

There is possibly a bigger structural issues on the page - do `Include failure scenarios` and `Treat unhandled requests` truly belong to `Request assertions`? After all, there nothing in them that relates to assertions... and that's why the confusion.

It would possibly make more sense if the whole page would be called `Testing Strategies` and the headings would be as follows:

```
# Testing Strategies
## Request assertions
### Special cases
## Include failure scenarios
## Treat unhandled requests
```

Let me know what you think.

**Comment by Izhaki** [comment]
I guess I'm waiting for feedback?

**Review by kettanaito — APPROVED** [review]
This is a good addition and can be merged as-is. Let's iterate on it if need be. Thank you, @Izhaki!

**Pull request merged**